extensions/mongodb-client/runtime/src/main/java/io/quarkus/mongodb/health/MongoHealthCheck.java

```diff
@@ -17,40 +17,68 @@
-            } else {
-                MongoClient client = (MongoClient) Arc.container().instance(bean.getName()).get();
-                clients.put(bean.getName(), client);
+        for (InstanceHandle<MongoClient> handle : Arc.container().select(MongoClient.class, Any.Literal.INSTANCE).handles()) {
+            String clientName = getMongoClientName(handle.getBean());
+            clients.put(clientName == null ? CLIENT_DEFAULT : clientName, handle.get());
+        }
+        // reactive clients
+        for (InstanceHandle<ReactiveMongoClient> handle : Arc.container()
+                .select(ReactiveMongoClient.class, Any.Literal.INSTANCE).handles()) {
+            String clientName = getMongoClientName(handle.getBean());
+            reactiveClients.put(clientName == null ? CLIENT_DEFAULT_REACTIVE : clientName, handle.get());
+        }
+    }
+
+    /**
+     * Get mongoClient name if defined.
+     *
+     * @param bean
+     * @return mongoClient name or null if not defined
+     * @see MongoClientName
+     */
+    private String getMongoClientName(Bean bean) {
+        for (Object qualifier : bean.getQualifiers()) {
+            if (qualifier instanceof MongoClientName) {
+                return ((MongoClientName) qualifier).value();
             }
         }
+        return null;
     }
 
     @Override
     public HealthCheckResponse call() {
         HealthCheckResponseBuilder builder = HealthCheckResponse.named("MongoDB connection health check").up();
+        Document command = new Document("ping", 1);
         for (Map.Entry<String, MongoClient> client : clients.entrySet()) {
-            boolean isDefault = DEFAULT_CLIENT.equals(client.getKey());
-            MongoClient mongoClient = client.getValue();
             try {
-                Document document = mongoClient.getDatabase("admin").runCommand(new Document("ping", 1));
-                String mongoClientName = isDefault ? "default" : client.getKey();
-                builder.up().withData(mongoClientName, document.toJson());
+                Document document = client.getValue().getDatabase("admin").runCommand(command);
+                builder.up().withData(client.getKey(), document.toJson());
+            } catch (Exception e) {
+                return builder.down().withData("reason", "client [" + client.getKey() + "]: " + e.getMessage()).build();
+            }
+        }
+        for (Map.Entry<String, ReactiveMongoClient> client : reactiveClients.entrySet()) {
+            try {
+                Document document = client.getValue().getDatabase("admin").runCommand(command).await().indefinitely();
+                builder.up().withData(client.getKey(), document.toJson());
             } catch (Exception e) {
-                return builder.down().withData("reason", e.getMessage()).build();
+                return builder.down().withData("reason", "client [" + client.getKey() + "]: " + e.getMessage()).build();
```

Comment:
Should it be `reactive client` or will it be clear from the name? I don't know if you can have both with the same name?

Reply:
it will be either <default-reactive> or the given client name.

Reply:
Good point! @lkrzyzanek may I ask you to update this line with 'reactive' to distinguish the two clients?

Reply:
done

Reply:
Also if you can register both with the same name, wondering if the `withData` will be overridden. I'll let @loicmathieu confirm if it's a problem or not.

Reply:
It's not possible today to register both synchronous and reactive client with the same name as both will have the same `@Named` annotation as we add it to the client.
On MP Health side, this wil generate two data with the same key so it would generates an invalid JSON.
I propose to keep it this way for now, on normal case, a user should use synchronous or reactive client not both (we may split the extension in two some day).

Reply:
> On MP Health side, this wil generate two data with the same key so it would generates an invalid JSON.

This is wrong as we produce one check by client so we are good.